srcpkgs/python3-jose/template

```diff
@@ -0,0 +1,20 @@
+# Template file for 'python3-jose'
+pkgname=python3-jose
+version=3.3.0
+revision=1
+wrksrc="python-jose-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+makedepends="python3-pip python3-rsa python3-ecdsa python3-pyasn1"
```

Comment:
why does it need pip?

Reply:
the install check for `pytest-runner`

```
/usr/lib/python3.10/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
  warnings.warn(
WARNING: The wheel package is not available.
/usr/bin/python3: No module named pip
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/setuptools/installer.py", line 82, in fetch_build_egg
    subprocess.check_call(cmd)
  File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/usr/bin/python3', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmp5wzuqyx1', '--quiet', 'pytest-runner']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/builddir/python-jose-3.3.0/setup.py", line 33, in <module>
    setup(
  File "/usr/lib/python3.10/site-packages/setuptools/__init__.py", line 86, in setup
    _install_setup_requires(attrs)
  File "/usr/lib/python3.10/site-packages/setuptools/__init__.py", line 80, in _install_setup_requires
    dist.fetch_build_eggs(dist.setup_requires)
  File "/usr/lib/python3.10/site-packages/setuptools/dist.py", line 876, in fetch_build_eggs
    resolved_dists = pkg_resources.working_set.resolve(
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 789, in resolve
    dist = best[req.key] = env.best_match(
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 1075, in best_match
    return self.obtain(req, installer)
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 1087, in obtain
    return installer(requirement)
  File "/usr/lib/python3.10/site-packages/setuptools/dist.py", line 956, in fetch_build_egg
    return fetch_build_egg(self, req)
  File "/usr/lib/python3.10/site-packages/setuptools/installer.py", line 84, in fetch_build_egg
    raise DistutilsError(str(e)) from e
distutils.errors.DistutilsError: Command '['/usr/bin/python3', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmp5wzuqyx1', '--quiet', 'pytest-runner']' returned non-zero exit status 1.
=> ERROR: python3-jose-3.3.0_1: do_build: 'python3 setup.py build ${make_build_args}' exited with 1
=> ERROR:   in do_build() at common/build-style/python3-module.sh:22
```

Reply:
based on what pytest-runner is (allowing `setup.py test` to call pytest), I think the best path forward would be to patch out that dependency in setup.py (it's gone on the HEAD of master), because the python3-module build style will run pytest directly if you add `python3-pytest` to `checkdepends`.